upgrade/upgrade-1.2.12.php

```diff
@@ -30,5 +32,42 @@
  */
 function upgrade_module_1_2_12($module)
 {
-    return $module->registerHook('displayInvoiceLegalFreeText');
+    foreach (PrestaShop\Module\PrestashopCheckout\OrderStates::ORDER_STATES as $state => $color) {
+        $orderStateId = (int) \Configuration::get($state);
```

Comment:
`$orderStateId = (int) \Configuration::getGlobalValue($state);`
OrderState are always global for all shops

Reply:
@Matt75 it's a copy paste of OrderState->setStateIcons, can you please check the description about this.

Reply:
`OrderStates::setStateIcons()` is updated in https://github.com/PrestaShopCorp/ps_checkout/pull/281 to use `\Configuration::getGlobalValue($state);` too

Reply:
ye ok then i should just change method scope and use it there to avoid maintaining it ?

Reply:
Maybe we can extract that to a Command to handle it properly ?
We could then execute it on OrderState class and upgrade file

Reply:
what do you think @v4lux, @apacios ?

Reply:
If it is just a copy/paste it could indeed be better to change the scope of the method setStateIcons().

Reply:
@Quetzacoalt91 can you look at this and above discussion and decide for us ? :D

Reply:
I'm in favor in reusing code when possible. In this case, the code is the same for adding or updating the order states.

## Solution 1

We could have the following upgrade:

```php
function upgrade_module_1_2_12($module)
{
    (new PrestaShop\Module\PrestashopCheckout\OrderStates)->updatePaypalStates();

    return $module->registerHook('displayInvoiceLegalFreeText');
}
```

And in OrderStates.php
```php
    /**
     * Update color for each ps_checkout order states
     *
     * @return bool
     */
    public function updatePaypalStates()
    {
        foreach (self::ORDER_STATES as $state => $color) {
            $orderStateId = $this->getPaypalStateId($state, $color);
            $this->setStateIcons($state, $orderStateId);
        }

        return true;
    }
```

## Solution 2

```php
function upgrade_module_1_2_12($module)
{
    (new PrestaShop\Module\PrestashopCheckout\OrderStates)->installPaypalStates();

    return $module->registerHook('displayInvoiceLegalFreeText');
}
```

By calling this method, we run the same code as during the installation process and make sure all order states exist before updating them. If they already exist on the shop, the code simply `continue` to the next order state.